docs/spire-multi-cluster.md

```diff
@@ -0,0 +1,314 @@
+```
+/tmp/space-x02/kubeconfig
+/tmp/kube-01/kubeconfig
+```
+
+Then using these individual files, create secrets in Tornjak cluster.
+_Note_: don't use special characters, stick to alphanumerics.
+
+```
+export CLUSTERNAME=...
+kubectl -n tornjak create secret generic $CLUSTERNAME --from-file=/tmp/$CLUSTERNAME/kubeconfig
+```
+so we get:
+```
+kubectl -n tornjak create secret generic space-x02 --from-file=/tmp/space-x02/kubeconfig
+kubectl -n tornjak create secret generic kube-01 --from-file=/tmp/kube-01/kubeconfig
+```
+
+## Step 2. Update SPIRE Server configuration.
+Update the SPIRE server configuration, by extending the NodeAttestor "k8s_psat"
+arguments.
+
+For all remote servers we need `kube_config_file` that contains its cluster KUBECONFIG.
+We don't need this for the local cluster.
+
+We assume all the SPIRE agent are deployed in `spire` namespace under
+`spire-agent` service-account, otherwise change the configuration accordingly:
+
+Update the spire-server ConfigMap:
+
+```console
+kubectl -n tornjak edit configmap spire-server
+```
+
+```yaml
+NodeAttestor "k8s_psat" {
+  plugin_data {
+      clusters = {
+          "space-x01" = {
+              service_account_whitelist = ["spire:spire-agent"]
+          },
+          "space-x02" = {
+              service_account_whitelist = ["spire:spire-agent"]
+              kube_config_file = "/tmp/space-x02/kubeconfig"
+          },
+          "kube-01" = {
+              service_account_whitelist = ["spire:spire-agent"]
+              kube_config_file = "/tmp/kube-01/kubeconfig"
+          },
+      }
+  }
+}
+```
+
+## Step 3. Modify SPIRE Server StatefulSet deployment
+Here we want to add the KUBECONFIG to be available to SPIRE server. We will mount
+the secrets created earlier to SPIRE server instance.
+Update the spire-server StatefulSet and look for `spire-server` container, then
+add the volume mounts, one per every remote cluster (secret)
+
+```console
+kubectl -n tornjak edit statefulset spire-server
+```
+
+This might look like this:
+```yaml
+...
+name: spire-server
+...
+volumeMounts:
+- mountPath: /tmp/space-x02
+  name: kubeconfig-space-x02
+- mountPath: /tmp/kube-01
+  name: kubeconfig-kube-01
+...
+volumes:
+- name: kubeconfig-space-x02
+  secret:
+    defaultMode: 420
+    secretName: space-x02
+- name: kubeconfig-kube-01
+  secret:
+    defaultMode: 420
+    secretName: kube-01
+```
+
+Then restart the SPIRE/Tornjak pod
+
+```console
+kubectl -n tornjak delete po spire-server-0
+```
+
+Verify if Tornjak/SPIRE server is running well, check the logs and make sure you
+can connect to Tornjak dashboard.
+
+## Step 4. Configure remote agents
+This work has to be done in every remote cluster.
+
+Gather from Tornjak server the `spire-bundle` that contains the certificates.
+
+```console
+kubectl -n tornjak get spire-bundle -oyaml > spire-bundle.yaml
+```
+Edit the file by changing the namespace from `tornjak` to `spire`
+and then update the `spire-bundle` on every remote cluster:
```

Comment:
maybe we can use kubectl patch to make this cleaner, something like

`
kubectl -n tornjak get spire-bundle -oyaml | k patch --type json --patch "[{"op": "replace", "path": "/metadata/namespace", "value":"tornjak"}]"  -f - --dry-run=client -oyaml
`